src/main/java/com/google/step/finscholar/servlets/ScholarshipListServlet.java

```diff
@@ -0,0 +1,80 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.finscholar.servlets;
+
+import static com.google.step.finscholar.data.ServletConstantValues.DEFAULT_VALUE;
+import static com.google.step.finscholar.data.ServletConstantValues.INVALID_INT_PARAMETER;
+import static com.google.step.finscholar.data.ServletConstantValues.JSON_CONTENT_TYPE;
+import static com.google.step.finscholar.data.ServletConstantValues.SCHOLARSHIP_COLLECTION_NAME;
+import static com.google.step.finscholar.data.ServletConstantValues.UNABLE_TO_LOAD_FIREBASE;
+import static com.google.step.finscholar.data.ServletConstantValues.UNABLE_TO_READ_FROM_FIRESTORE;
+import static com.google.step.finscholar.data.ServletConstantValues.UNKNOWN_ERROR;
+
+import static com.google.step.finscholar.data.Utils.getIntParameter;
+import static com.google.step.finscholar.data.Utils.getStringParameter;
+import static com.google.step.finscholar.firebase.FirebaseStorageManager.getCollectionBatch;
+
+import com.google.step.finscholar.firebase.FirebaseAppManager;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.FirebaseException;
+import com.google.firebase.cloud.FirestoreClient;
+import java.io.IOException;
+import java.lang.NumberFormatException;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/scholarship-list")
+public class ScholarshipListServlet extends HttpServlet {
+
+  private static String ITEMS_PER_BATCH = "numberOfItems";
+  private static String ID_OF_LAST_ITEM = "idOfLastItem";
+  private static String SORT_BY = "sortBy";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Firestore database = null;
+    int itemsPerBatch = 0;
+    try {
+      database = FirestoreClient.getFirestore(FirebaseAppManager.getApp());
+      itemsPerBatch = getIntParameter(request, ITEMS_PER_BATCH);
+    } catch (NumberFormatException numberException) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, 
+          INVALID_INT_PARAMETER + numberException);
+    } catch (Exception e) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, 
+          UNKNOWN_ERROR + e);
+    }
+
+    if (database != null && itemsPerBatch != 0) {
+      try {
+        String idOfLastItem = getStringParameter(request, ID_OF_LAST_ITEM, DEFAULT_VALUE);
+        String sortBy = getStringParameter(request, SORT_BY, DEFAULT_VALUE);
```

Comment:
I can't remember where else I commented about this, but I don't think it makes sense to set these to something as arbitrary as an empty string. I would just leave them as Optionals and pass those to the getCollectionBatch. This is less error prone and protects against NPEs.